promutils/labeled/counter.go

```diff
@@ -73,15 +61,19 @@ func NewCounter(name, description string, scope promutils.Scope, opts ...MetricO
 		if _, emitUnlabeledMetric := opt.(EmitUnlabeledMetricOption); emitUnlabeledMetric {
 			c.Counter = scope.MustNewCounter(GetUnlabeledMetricName(name), description)
 		} else if additionalLabels, casted := opt.(AdditionalLabelsOption); casted {
-			labels := GetUniqueLabels(metricStringKeys, additionalLabels.Labels)
-			// Here we only append the labels that don't exist in metricStringKeys
-			c.CounterVec = scope.MustNewCounterVec(name, description, append(metricStringKeys, labels...)...)
-			c.additionalLabels = contextutils.MetricKeysFromStrings(labels)
+			// compute unique labels
+			labelSet := sets.NewString(metricStringKeys...)
+			labelSet.Insert(additionalLabels.Labels...)
+			labels := labelSet.List()
```

Comment:
This might not reserve the order of labels. I hope that doesn't matter.

Reply:
Internally these labels [are sorted](https://github.com/prometheus/client_golang/blob/2c3d072cdd4dbd9dbbfe415cbd01851db8962289/prometheus/desc.go#L106) before stored, so this shouldn't be an issue. If it ever changes, it will break the unit new unit tests so I think we are OK here.

Reply:
Great! Thanks for the pointer.